src/core/aio/disk_engine.h

```diff
@@ -68,14 +67,12 @@ class disk_file
     work_queue<aio_task> _read_queue;
 };
 
-class disk_engine
+class disk_engine : public utils::singleton<disk_engine>
 {
 public:
-    disk_engine(service_node *node);
+    disk_engine();
```

Comment:
better to move ctor and dctor to private section, to restrict create and delete it freely.

Reply:
We can't make ctor and dctor to be private member, because we derived from `singleton`, to implement sigleton. And disk_engine is created in `singleton::instance`